### PR TITLE
Add kubectl trace v0.1.0-rc.1

### PR DESCRIPTION
**Pull request by fntlnz**
Happy to submit kubectl trace to the krew index as per the discussions happening one year ago  in https://github.com/iovisor/kubectl-trace/issues/17

Sorry if it took a long time!

Adding a wip since I didn't test the manifest on all platforms yet.

If anyone can test it on windows before I do and report here I'll be grateful.

Signed-off-by: Lorenzo Fontana <lo@linux.com>

### CODE REVIEW

File: plugins/trace.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# limitations under the License.
+
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: trace
+spec:
+  version: "v0.3.4"
+  homepage: https://github.com/iovisor/kubectl-trace
+  shortDescription: Schedule bpftrace programs on your kubernetes cluster using the kubectl.
```
Comment (ahmetb) [inline plugins/trace.yaml:22]: `Schedule bpftrace programs on your cluster`

would be a short summary that doesn't get truncated.

File: plugins/trace.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  caveats: |
+    kubectl trace is now installed!
```
Comment (ahmetb) [inline plugins/trace.yaml:24]: we usually reserve `caveats` for software that needs to be installed after plugin is installed.

you should probably already have the following examples in your `--help`?

**Comment by ahmetb** [comment]
> If anyone can test it on windows before I do and report here I'll be grateful.

I suspect neither of the maintainers have a windows box. :(

File: plugins/trace.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+metadata:
+  name: trace
+spec:
+  version: "v0.3.4"
+  homepage: https://github.com/iovisor/kubectl-trace
+  shortDescription: Schedule bpftrace programs on your kubernetes cluster using the kubectl.
+  caveats: |
+    kubectl trace is now installed! 
+
+    Examples:
```
Comment (corneliusweig) [inline plugins/trace.yaml:26]: I think these examples would fit better in a `description`, of as @.ahmetb said in the `--help` output.

Also, it's usually nice to put a rough description of the technical details into the description. Just to give the users an idea of what is going on when invoking the plugin.

Comment (fntlnz) [reply plugins/trace.yaml:26]: That's exactly what I wanted thanks @corneliusweig !

File: plugins/trace.yaml (hunk 2 above)
Comment (fntlnz) [reply plugins/trace.yaml:24]: I added them to caveats as suggetsed by @corneliusweig

File: plugins/trace.yaml (hunk 1 above)
Comment (fntlnz) [reply plugins/trace.yaml:22]: Thanks, will change this

**Comment by fntlnz** [comment]
@ahmetb @corneliusweig I just addressed all the suggestions and tested again.

Thank you two for the reviews, I really appreciate it. And thanks for this awesome tool!

**Comment by fntlnz** [comment]
Created the corresponding PR on kubectl trace for when this is merged. https://github.com/iovisor/kubectl-trace/pull/118

File: plugins/trace.yaml (hunk 3 above)
Comment (corneliusweig) [reply plugins/trace.yaml:26]: I think you gave a nice collection of examples which are well suited for people familiar with `kubectl-trace`. However, I'd recommend to keep the description on a more conceptual level and give the (potential) users an idea of the key concepts of `kubectl-trace`. Also, you should try to explain why and in which situations users could benefit from this plugin.
Further, you should explain how `trace` does its magic, for example:

> `trace` loads bpf programs into the host kernels on the target node and uses these programs to collect metrics.

(not sure if that's totally accurate, but you get the point, right?)

In essence, the description is a bit of a sales pitch for potential users. So use it wisely :)

**Comment by fntlnz** [comment]
Thanks for your suggestions @corneliusweig - updated!

**Comment by ahmetb** [comment]
It seems `description` now only lists examples. Was that your intention @fntlnz? If so, I'll go ahead and merge. BTW you can set up an easy [release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) to your repo so you don't have to come here to edit your yaml file every time you  have a new version.

**Comment by fntlnz** [comment]
Thanks everyone!

**Pull request merged**